internal/provider/data_source_docker_image_test.go

```diff
@@ -0,0 +1,162 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+var imageDigestRegexp = regexp.MustCompile(`^sha256:[A-Fa-f0-9]+$`)
+var imageNameWithTagAndDigestRegexp = regexp.MustCompile(`^.*:.*@sha256:[A-Fa-f0-9]+$`)
+
+func TestAccDockerImageDataSource_withSpecificTag(t *testing.T) {
```

Comment:
Is it possible to also test references to non-existent image/tag/sha256?

Reply:
yes, I can add such a test

Reply:
done in 86f24ca6153a783fe77ae3bcd160ce70efe463c9